legend-engine-xt-nonrelationalStore-mongodb-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/mongodb/plugin/MongoDBExecutionNodeExecutor.java

```diff
@@ -0,0 +1,205 @@
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.SequenceExecutionNode;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.graphFetch.GraphFetchExecutionNode;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.graphFetch.LocalGraphFetchExecutionNode;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.graphFetch.StoreMappingGlobalGraphFetchExecutionNode;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.graphFetch.store.inMemory.InMemoryCrossStoreGraphFetchExecutionNode;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.graphFetch.store.inMemory.InMemoryPropertyGraphFetchExecutionNode;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.graphFetch.store.inMemory.InMemoryRootGraphFetchExecutionNode;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.graphFetch.store.inMemory.StoreStreamReadingExecutionNode;
+import org.pac4j.core.profile.CommonProfile;
+
+import java.io.IOException;
+
+public class MongoDBExecutionNodeExecutor implements ExecutionNodeVisitor<Result>
+{
+    MutableList<CommonProfile> profiles;
+    ExecutionState executionState;
+
+    public MongoDBExecutionNodeExecutor(MutableList<CommonProfile> profiles, ExecutionState executionState)
+    {
+        this.profiles = profiles;
+        this.executionState = executionState;
+    }
+
+    @Override
+    public Result visit(ExecutionNode executionNode)
+    {
+        if (executionNode instanceof MongoDBExecutionNode)
+        {
+            try (Scope scope = GlobalTracer.get().buildSpan("MongoDB Store Execution").startActive(true))
+            {
+                return executeMongoDBExecutionNode((MongoDBExecutionNode) executionNode);
+            }
+        }
+        else
+        {
+            return null;
+        }
+    }
+
+    public Result executeMongoDBExecutionNode(MongoDBExecutionNode mongoDBExecutionNode)
+    {
+        try (Scope scope = GlobalTracer.get().buildSpan("MongoDB Store Execution").startActive(true))
+        {
+            scope.span().setTag("databaseCommand", (mongoDBExecutionNode).databaseCommand.toString());
+            MongoDBConnection mongoDBConnection = mongoDBExecutionNode.connection;
+            String databaseCommand = mongoDBExecutionNode.databaseCommand;
+
+            ObjectMapper objectMapper = new ObjectMapper();
+            DatabaseCommand dbCommand = objectMapper.readValue(databaseCommand, DatabaseCommand.class);
+            MongoDBQueryJsonComposer mongoDBQueryJsonComposer = new MongoDBQueryJsonComposer(false);
+            String composedDbCommand = mongoDBQueryJsonComposer.parseDatabaseCommand(dbCommand);
+
+            CredentialProviderProvider credentialProviderProvider = ((MongoDBStoreExecutionState) executionState.getStoreExecutionState(StoreType.NonRelational_MongoDB)).getCredentialProviderProvider();
```

Comment:
Fixed it, didn't realise that execution state had the credential provider - am guessing it will have the crednential from the connection/runtime object passed in.